src/storage/swiftOS.py

```diff
@@ -275,10 +311,14 @@ def delete_object_after(self,containername,prefix,objectname,ttl):
             logging.error(e)
             return -1
 
-    def delete_object(self,containername,prefix,objectname):
+    def delete_object(self,containername,prefix,objectname=None):
 
-        swift_account_url = "{swift_url}/{containername}/{prefix}/{objectname}".format(
-            swift_url=self.url, containername=containername, prefix=prefix, objectname=objectname)
+        if objectname!=None:
```

Comment:
Since now objectname can be `None`, I would suggest to change line `logging.info("Deleting object: {}/{}/{}".format(containername,prefix,objectname))` to `logging.info(f"Deleting {swift_account_url}")` or something similar, like you do for `s3v4OS`.

Reply:
Removed since `objectname` is not optional anymore in this function